newm-server/src/main/kotlin/io/newm/server/features/cardano/model/Key.kt

```diff
@@ -1,47 +1,76 @@
-package io.newm.chain.database.entity
+package io.newm.server.features.cardano.model
 
 import com.google.iot.cbor.CborArray
 import com.google.iot.cbor.CborByteString
 import com.google.iot.cbor.CborInteger
-import com.google.iot.cbor.CborReader
-import io.newm.chain.config.Config
-import io.newm.chain.model.CliKey
-import io.newm.chain.model.CliKeyPair
 import io.newm.chain.util.Bech32
 import io.newm.chain.util.Blake2b
 import io.newm.chain.util.Constants.PAYMENT_ADDRESS_PREFIX_MAINNET
 import io.newm.chain.util.Constants.PAYMENT_ADDRESS_PREFIX_TESTNET
 import io.newm.chain.util.Constants.PAYMENT_ADDRESS_SCRIPT_PREFIX_MAINNET
 import io.newm.chain.util.Constants.PAYMENT_ADDRESS_SCRIPT_PREFIX_TESTNET
-import io.newm.chain.util.hexToByteArray
 import io.newm.chain.util.toHexString
+import io.newm.server.features.cardano.repo.CardanoRepository
+import io.newm.shared.koin.inject
+import io.newm.shared.serialization.LocalDateTimeSerializer
+import io.newm.shared.serialization.UUIDSerializer
+import kotlinx.serialization.Serializable
 import org.bouncycastle.crypto.generators.Ed25519KeyPairGenerator
 import org.bouncycastle.crypto.params.Ed25519KeyGenerationParameters
 import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
 import org.bouncycastle.crypto.params.Ed25519PublicKeyParameters
 import java.security.SecureRandom
+import java.time.LocalDateTime
+import java.util.*
 import kotlin.concurrent.getOrSet
 
+@Serializable
 data class Key(
-    val id: Long? = null,
+    @Serializable(with = UUIDSerializer::class)
+    val id: UUID? = null,
+    @Serializable(with = LocalDateTimeSerializer::class)
+    val createdAt: LocalDateTime? = null,
     val skey: ByteArray,
-    val vkey: String,
+    val vkey: ByteArray,
     val address: String,
     val script: String?,
     val scriptAddress: String?,
-    val created: Long,
 ) {
+    override fun equals(other: Any?): Boolean {
```

Comment:
Why override equals and hashCode on this data class?

Reply:
I got a warning about overriding them because the ByteArrays wouldn't compare their contents otherwise.

Reply:
![image](https://user-images.githubusercontent.com/245918/229202423-b0603c44-c4f6-4637-9e16-3f77cd88dc42.png)